lib/rouge/lexers/sql.rb

```diff
@@ -20,7 +20,7 @@ def self.keywords
           CASE CAST CATALOG CATALOG_NAME CHAIN CHARACTERISTICS
           CHARACTER_LENGTH CHARACTER_SET_CATALOG CHARACTER_SET_NAME
           CHARACTER_SET_SCHEMA CHAR_LENGTH CHECK CHECKED CHECKPOINT
-          CLASS CLASS_ORIGIN CLOB CLOSE CLUSTER COALSECE COBOL COLLATE
+          CLASS CLASS_ORIGIN CLOB CLOSE CLUSTER COALSECE COBOL
```

Comment:
I gather these have been removed because they're proprietary Transact-SQL keywords? I can see the logic to that but it's a change beyond the scope of adding Hive. If you'd like to do that, can you submit that as a separate PR?

Reply:
No, that's not the reason. As part of this branch, I added support for distinguishing keywords from types (because `Keyword` and `Keyword::Type` are different highlight classes). In doing that, I moved some keywords.

For this specific one: `COLLATE` is both a function and part of a string type specification (https://dev.mysql.com/doc/refman/5.7/en/string-type-overview.html); I had to make a choice how to highlight. Happy to make the other choice as well. What do you think?

Reply:
See https://github.com/rouge-ruby/rouge/pull/880/commits/f3311f1ee814d045deb30208c9052ba36779804b

Reply:
I'll readily admit I have only cursory experience writing SQL but my natural instinct would not be to characterise something like `COLLATE` as a type. Do you have an opinion @vidarh?

Reply:
Yeah, I get where you're coming from. To be pricise, `COLLATE` is not a type but the string `TEXT CHARACTER SET latin1 COLLATE latin1_general_cs` is.

The "actual" solution would be not to make a choice but to just parse and see which one of these is it. I'd see that as a possible future improvement.

Reply:
In a silly way, I even like keeping it as-is as it might encourage someone who cares to make that parsing improvement in the future ;)

Reply:
@tkluck-booking I've had a look at what [Pygments](https://bitbucket.org/birkenfeld/pygments-main/src/default/pygments/lexers/sql.py) and [Chroma](https://github.com/alecthomas/chroma/blob/master/lexers/s/sql.go) do and while their approaches aren't dispositive, we do often look to them when trying to determine how best to solve a problem.

Both keep `collate` in the category of words classified as `Keyword` but both use the `Name::Builtin` token for the 'types' rather than `Keyword::Type`. That seems like a good compromise to me.

Could you move `collate` and the other keywords back and make that change?

Reply:
Done!

Reply:
@tkluck-booking I noticed you've moved some of those keywords back but not all. Was the rationale for the ones you left (eg. `YEAR`) that they weren't both functions and type 'attributes' (for want of a better word)?

Reply:
Sorry, should have added an explicit link to the commit message: https://github.com/rouge-ruby/rouge/pull/880/commits/32f5650b1ea0c7751b0a420b35d105464ba1fc7c

Reply:
My oversight—your explanation makes sense. Thanks!